core/column_descriptors.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  IColumnsDescriptor,
+  IRecordDescriptor,
+  IRecordDescriptorFields
+} from "@dataform/core/common";
+import * as utils from "@dataform/core/utils";
+import { dataform } from "@dataform/protos";
+
+/**
+ * @hidden
+ */
+export class ColumnDescriptors {
+  public static mapToColumnProtoArray(columns: IColumnsDescriptor): dataform.IColumnDescriptor[] {
+    return utils.flatten(
+      Object.keys(columns).map(column =>
+        ColumnDescriptors.mapColumnDescriptionToProto([column], columns[column])
+      )
+    );
+  }
+
+  public static mapColumnDescriptionToProto(
+    currentPath: string[],
+    description: string | IRecordDescriptor
+  ): dataform.IColumnDescriptor[] {
+    if (typeof description === "string") {
+      return [
+        dataform.ColumnDescriptor.create({
+          description,
+          path: currentPath
+        })
+      ];
+    }
+    const extraFields = Object.keys(description).filter(
+      key => !IRecordDescriptorFields.includes(key)
+    );
+    if (extraFields.length > 0) {
+      throw new Error(
+        `Unknown property "${extraFields[0]}" in column descriptor. Allowed properties are: ${IRecordDescriptorFields}`
```

Comment:
Dan and I just agreed that we wouldn't fix this... It feels quite un-JavaScripty to do this, are you sure? I guess I'm happy if you are, though if we are going to do it I think we should be consistent with all `config` fields, as it'll be confusing if users start to expect us to do that and we don't

Reply:
I'm happy to take this all the way... Having just written some configs, getting no feedback about if I'm doing it right I found very frustrating, particularly without good documentation for this alpha. 
We definitely get a fair few support requests where people have typos. We could downgrade the errors to not fully throw, but put them in the graph instead.

Reply:
Have taken it all the way